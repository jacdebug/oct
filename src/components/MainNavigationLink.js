import React from 'react';
import styled, { css } from 'styled-components';
import { NavLink } from 'react-router-dom';

const MainNavigationLink = styled(({ bright, ...rest }) =>
  <NavLink {...rest} />
)`
   padding: 8px 15px;
   text-decoration: none;
   font-size: 13px;

   ${props =>
     props.bright &&
     css`
      color: ${props => props.theme.tcolor4};
  `}
`;

export default MainNavigationLink;
