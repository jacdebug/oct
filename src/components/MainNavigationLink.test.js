import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MainNavigationLink from './MainNavigationLink';

const linkTestData = {
  name: 'name1',
  url: '/',
};

it('renders without crashing', () => {
  shallow(
    <MainNavigationLink to={linkTestData.url}>
      {linkTestData.name}
    </MainNavigationLink>
  );
});

it('renders a snapshot', () => {
  const tree = shallow(
    <MainNavigationLink to={linkTestData.url}>
      {linkTestData.name}
    </MainNavigationLink>
  );
  expect(toJson(tree)).toMatchSnapshot();
});
