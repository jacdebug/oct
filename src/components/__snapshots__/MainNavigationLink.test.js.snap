// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<Component
  className="sc-bdVaJa cOhLSd"
  to="/"
>
  name1
</Component>
`;
